server/routes/charts.js

```diff
@@ -189,4 +189,28 @@ router
       return res.status(500).json({ message: error.message })
     }
   })
+  .patch(ensureAuthenticated, async (req, res) => {
```

Comment:
should this be a `.post`? just thinking we're not updating an existing record here, we're creating a  new one

Reply:
Hmm backing up for a sec, we may want this handler on a different route altogether. You're right PUT and PATCH are both Edit actions for a route (I think PUT isn't update or create, but an update via wholesale overwrite of an existing doc _(you need to send everything, including attributes you don't want to change, whereas PATCH is just updating a few things, and you only need send those attributes)._
But in this case we're not editing an existing chart, we're creating a duplicate so maybe we want something like the following?
```js
router
  .route('/api/v1/charts/:chart_id/duplicate') 
  .post(//...
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/127/commits/b5f38cb418e1f1c0da0dcafb7354b98d8c00c1d0